### PR TITLE
Http framework independent

### PR DESCRIPTION
**Pull request by trygve-lie**
This PR makes the Podlet http framework free making it possible to integrate the Podlet with multiple http frameworks such as Hapi, Fastify etc.

Previously Podium has been very tied to Express.js where each module in the request chain has been a middleware where properties from each module has been written to `res.locals.podium`. Internally multiple of Express.js only methods have also been used making Podium very tied to Express.js.

This implementation switches from using a middleware chain passing request and response objects around to a Promise chain where a HttpIncoming object we have controll over is passed on. In other words; each module in Podium does now have a `.process()` method which can be chained together making it possible for any other http framwork to be integrated.

All usage of Express.js only methods has also been replaced with generic methods which should work within any http framework.

Currently there is a Hapi plugin using this: https://github.com/podium-lib/hapi-podlet

Despite making Podium http framework free, we treat Connect middleware based http frameworks (such as Express.js) first class by keeping the already existing `.middleware()` method. This method does now just wrap the `.process()` method. Due to this its easy to get started while it also provide backward compabillity.

### CODE REVIEW

File: lib/podlet.js [hunk 1]
```diff
@@ -316,95 +288,85 @@ const PodiumPodlet = class PodiumPodlet {
...
-            return utils.uriIsRelative(uri)
-                ? utils.pathnameBuilder(pathname, uri)
-                : uri;
+    async process(incoming) {
+        incoming.view = this._view;
+        incoming.name = this.name;
+        incoming.css = this.cssRoute;
+        incoming.js = this.jsRoute;
+
+        // Determin if request comes from layout server or not
```
Comment (omhoumz) [inline lib/podlet.js]:
```suggestion
        // Determine if request comes from layout server or not
```

Comment (trygve-lie) [reply lib/podlet.js]: @omhoumz Thnx :smiley:

File: lib/podlet.js [hunk 2]
```diff
@@ -259,28 +231,28 @@ const PodiumPodlet = class PodiumPodlet {
...
-                `One can not define more than 4 proxy targets for each podlet`,
+                `One can not define more than 4 proxy targets for each podlet`
             );
         }
 
         this.proxyRoutes[name] = target;
 
         if (this.development) {
-            this._proxy.register(this);
+            this.httpProxy.register(this);
```
Comment (digitalsadhu) [inline lib/podlet.js]: I saw that one of the examples is referring to `this._proxy` is that correct or does it need to be updated?

Comment (trygve-lie) [reply lib/podlet.js]: No. The example you refere to are more tests on implementing misc http frameworks. Will do those into proper modules and then delete the example.

**Pull request merged**